core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -763,11 +763,19 @@ private static AuthSession fromTokenResponse(
         long startTimeMillis,
         AuthSession parent,
         String credential) {
+      // issued token type is not present in every OAuth2 response:
+      // assume type is access token if none provided.
+      // See https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.3
```

Comment:
@adutra do you maybe know if Keycloak fully supports RFC 8693? The token exchange follows https://www.rfc-editor.org/rfc/rfc8693#name-successful-response, where `issued_token_type` is `required`

Reply:
I found https://github.com/keycloak/keycloak/discussions/26502, which states unfortunately that Keycloak deviates from the standard

Reply:
Indeed, Keycloak does not fully implement it. Quoting from their [Securing Applications and Services Guide](https://www.keycloak.org/docs/latest/securing_apps/index.html#how-token-exchange-works): 

> Token exchange in Keycloak is a very loose implementation of the OAuth Token Exchange specification at the IETF. We have extended it a little, ignored some of it, and loosely interpreted other parts of the specification. It is a simple grant type invocation on a realm’s OpenID Connect token endpoint.

Reply:
But the problem here is wider: `issued_token_type` is only defined for the token exchange flow (RFC 8693). This field does not exist for standard flows from RFC 6749. So the following scenario can happen even with fully-compliant servers:

1. client uses `client_credentials` to authenticate initially;
2. server sends a successful response similar to [this one](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.3); note that it does not contain `issued_token_type`, which is normal since it's not defined for this flow.
3. client attempts to refresh the token using the token exchange flow;
4. client throws IAE because `subjectTokenType` is null.

Reply:
@nastra also see this comment which gives some context on why we have the problem today: https://github.com/apache/iceberg/pull/4771#discussion_r874114263